apps/vmq_server/src/vmq_reg_mgr.erl

```diff
@@ -162,15 +162,14 @@ setup_queue(SubscriberId, Subs, Acc) ->
 
 handle_event(Handler, Event) ->
     case Handler(Event) of
-        {delete, _, _} ->
+        {delete, _SubscriberId, _} ->
             %% TODO: we might consider a queue cleanup here.
             ignore;
         {update, _SubscriberId, [], []} ->
             %% noop
             ignore;
-        {update, SubscriberId, _, _} ->
-            Subs = vmq_reg:subscriptions_for_subscriber_id(SubscriberId),
-            handle_new_sub_event(SubscriberId, Subs);
+        {update, SubscriberId, _OldSubs, NewSubs} ->
```

Comment:
Does this work? The `_OldSubs` and `NewSubs` don't contain the 'real' DB object, they contain the changes (from `vmq_subscriber:get_changes`) made to them, e.g. what was added or what was removed... Or in other words the `NewSubs` isn't identical to what you have in `Subs`.

Reply:
Ah, missed the fact that you've removed the call to `get_changed` ;)